core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -57,7 +57,7 @@ private[spark] class HttpServer(
     } else {
       logInfo("Starting HTTP Server")
       val (actualServer, actualPort) =
-        Utils.startServiceOnPort[Server](requestedPort, doStart, serverName)
+        Utils.startServiceOnPort[Server](requestedPort, doStart, new SparkConf(), serverName)
```

Comment:
We should pass this a conf that already exists instead of creating a new one. This requires changing the constructors of `HttpServer` and `HttpFileServer`. I believe there is already a conf everywhere else.